eng/references.targets

```
@@ -94,6 +85,16 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="ReplaceCoreLibSrcWithRefAssemblyForCompilation"
+          AfterTargets="FindReferenceAssembliesForReferences"
+          Condition="'@(_coreLibProjectReference)' != ''">
```

nit: this target should only run when not compiling against reference assemblies.

```suggestion
          Condition="'@(_coreLibProjectReference)' != '' and '$(CompileUsingReferenceAssemblies)' != 'true'">
```